src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -175,20 +180,20 @@ uct_ib_mlx5_devx_reg_ksm_data_mt(uct_ib_mlx5_md_t *md, void *address,
     ucs_carray_for_each(mr, ksm_data->mrs, ksm_data->mr_num) {
         uct_ib_mlx5_devx_klm_entry_set(&klm, mr - ksm_data->mrs, mr_address,
                                        *mr);
-        mr_address = UCS_PTR_BYTE_OFFSET(mr_address, (*mr)->length);
+        mr_address = UCS_PTR_BYTE_OFFSET(mr_address, chunk_size);
     }
 
-    if ((void*)iova != address) {
+    if (ksm_iova + length > (uintptr_t)mr_address) {
```

Comment:
why need to add length?

Reply:
why can't we compare  iova!=address like before?

Reply:
Reverted.